staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere_test.go

```diff
@@ -205,6 +205,95 @@ func configFromEnvOrSim() (VSphereConfig, func()) {
 	return configFromSim()
 }
 
+func TestSecretUpdated(t *testing.T) {
+	datacenter := "0.0.0.0"
+	secretName := "vccreds"
+	secretNamespace := "kube-system"
+	username := "test-username"
+	password := "test-password"
+	basicSecret := fakeSecret(secretName, secretNamespace, datacenter, username, password)
+
```

Comment:
Could you tell me what are difference from original func TestSecretUpdated(t *testing.T) ？Thanks!

Reply:
the TestSecretUpdated function is adjusted to avoid being affected by other test methods, thus avoiding `DATA RACE`.

Just adjusted the order.

Reply:
Generally the testing package is going to run tests in parallel (unless you explicitly tell it to run the tests serially or play some form of game with locking). If you are running in parallel then my guess is you haven't actually avoided the ```DATA RACE``` though you may have made it less likely.

Reply:
tests inside one _test.go file are running in parallel only if you add t.Parallel() and run with -parallel tag.
by default the command go test runs in parallel tests for different packages, and default is the number of CPUs available (see go help build)